src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5465,6 +5470,17 @@ int wm_vuldet_collect_agents_to_scan(wm_vuldet_t *vuldet) {
             os_strdup(agti_arch, agents->arch);
         }
 
+        // Correcting the achitecture string to generate a valid OS CPE
+        if(agents->arch) {
```

Comment:
We should make sure that this architecture translation doesn't affect the Linux agents scans. Not sure whether this architecture string is also used in the Linux agents scans.
The string will match for sure with the strings used by the NVD, but not sure which strings are using in the OVAL feeds.

Reply:
The team has been testing this, and the results indicate that the change in the architecture is an improvement in the detection.